### ISSUE TITLE
XML validation output is too verbose

### ISSUE THREAD

**Issue by thomasleplus** [super-linter/super-linter#6653]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Are you using the latest Super-linter version available?

- [x] I am using the latest Super-linter version.
- [x] I can reproduce the issue running Super-linter using complete version identifier (example: vX.Y.Z), and not just with a partial one (example: vX)

- [x] I am using the super-linter/super-linter action or container image, and not the deprecated github/super-linter action or container image.

### Are you resonably sure that it's a Super-linter issue, and not an issue related to a tool that Super-linter runs?

- [x] I think that this is a Super-linter issue.

### Current Behavior

Hi,

I am running the image `super-linter/super-linter:latest` and I noticed for a while that the output of the XML validation tool is very verbose. Basically it seems to output the whole content of the validated XML files even when no issue is found. This is flooding my CI/CD logs and I think that it could also be a security concern if any of my XML files contained some sensitive data that I don't want to leak into the CI/CD logs.

The output looks like this in my logs:

```text
2025-03-10 03:05:30 [INFO]   Linting XML items...
2025-03-10 03:05:30 [NOTICE]   XML linted successfully
2025-03-10 03:05:30 [INFO]   Command output for XML:
------
<?xml version="1.0" encoding="UTF-8"?>
...
```

Please let me know if there is an option to disable this behavior? Otherwise I think there should be such an option and it should probably be disabled by default IMHO.

Cheers,

Thomas

### Expected Behavior

When no issue is found, the log should just say:

```text
2025-03-10 03:05:30 [INFO]   Linting XML items...
2025-03-10 03:05:30 [NOTICE]   XML linted successfully
2025-03-10 03:05:31 [INFO]   Linting YAML items...
```

### Super-Linter version

```markdown
v7.3.0
```

### Super-linter configuration

```yaml
jobs:
  lint:
    name: Lint Code Base
    runs-on: ubuntu-latest
    permissions:
      contents: read
      packages: read
      statuses: write
    steps:
      - name: Checkout Code
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
        with:
          fetch-depth: 0
      - name: Lint Code Base
        uses: super-linter/super-linter@4e8a7c2bf106c4c766c816b35ec612638dc9b6b2 # v7.3.0
        env:
          VALIDATE_ALL_CODEBASE: true
          LINTER_RULES_PATH: .
          DEFAULT_BRANCH: main
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

### Relevant log output

```shell
2025-03-10 03:05:30 [INFO]   Linting XML items...
2025-03-10 03:05:30 [NOTICE]   XML linted successfully
2025-03-10 03:05:30 [INFO]   Command output for XML:
------
<?xml version="1.0" encoding="UTF-8"?>
```

### Steps To Reproduce

Use the GitHub Actions configuration above on any GitHub repo containing XML files.

**Comment by thomasleplus** [comment]
I think that the `-noout` option of `xmllint` could be the solution here. I am happy to create a PR if this approach is accepted.

**Comment by ferrarimarco** [comment]
Hi @thomasleplus. I think that would be fine. Thanks.

**Comment by thomasleplus** [comment]
Great. I have the PR almost ready, just need some testing.

I see 3 options:
1. echoing is disabled, no option to turn it back on.
2. echoing is disabled by default but can be turned on.
3. echoing is enabled by default (same as today) but can be turned off.

I tend to prefer option 2 because I think that it is the default behavior that most users would want/expect. Option 3 as the least chance of regression (e.g. if someone actually relies on the XML content being echoed which sounds unlikely/kludgy) but then other users might never know that they can disable the echoing.

What do you think?

**Comment by ferrarimarco** [comment]
(by the way, I'd consider this an enhancement, rather than a bug, because this is not broken.)

I'd say that option 2 is ok, but I'm not sure we should tie it to the `LOG_LEVEL` variable, because that is more for debugging Super-linter itself, although we did it for at least another linter:

https://github.com/super-linter/super-linter/blob/4e1eb5f5e0bffb6484d6411426a4edd32e35eeed/lib/linter.sh#L184-L187

I suppose we can go ahead as you propose in #6654, and eventually add a variable to get xmllint output if there's demand for that.

Thanks!

**Comment by thomasleplus** [comment]
I've made the requested changes and moved #6654 from draft to open status.

Thank you for your feedback.
